mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -52,7 +52,7 @@ public AccountManagerUtil(Context context, String accountName, boolean sharedSto
             //Attempt to retrieve the account
             Account[] accounts = mAccountManager.getAccountsByType(accountType);
             for (Account account : accounts) {
-                if (accountName.equals(account.name)) {
+                if (accountName != null && accountName.equals(account.name)) {
                     String password = mAccountManager.getPassword(account);
                     String savedPassword = identifier.toString();
                     if (password.equals(savedPassword)) {
```

Comment:
Add null check to password also as per logs here
https://github.com/CAAPIM/Android-MAS-SDK/issues/340